packages/macro/index.d.ts

```diff
@@ -7,15 +7,16 @@ export function t(
   ...placeholders: any[]
 ): string
 
+export type UnderscoreDigit<T = string> = { [digit: string]: T }
```

Comment:
Isn't the problem after all that previously the `digit` was a `number`, now it's a `string`? That would make probably sense, because all object keys are strings (or normalized to strings).

Reply:
We can't do this, because T as string is not assignable to offset as number

```
export type ChoiceOptions<T = string> = {
	offset?: number;
	zero?: T;
	one?: T;
	few?: T;
	many?: T;
	other?: T;
	[digit: string]: T;
};
```

Reply:
and with this: 
```
export type ChoiceOptions<T = string> = {
	offset?: number;
	zero?: T;
	one?: T;
	few?: T;
	many?: T;
	other?: T;
	[digit: number]: T;
};

const Plural: React.ComponentType<ChoiceProps> = (props) => {
	return <div>{props["_0"]}</div>;
};

class App extends React.Component {
	render() {
		return <Plural _0="one" />;
	}
}
```

_0 fails because typescript interprets it's a string

Reply:
https://codesandbox.io/s/react-typescript-playground-forked-becf6?file=/src/index.tsx:463-645

Reply:
Got it, thank you!